src/plugin/eosTransaction.ts

```diff
@@ -464,62 +463,56 @@ export class EosTransaction implements Interfaces.Transaction {
-    const uniqueRequiredAuths = await this.addAuthToPermissions(requiredAuthsArray)
+    const uniqueRequiredAuths = await this.addAuthToPermissions(requiredAuthsUniqueArray)
     // Attach subAuth for each accountName/permission specified as Authorization.
     const uniqueRequiredAuthsWithSubAuthPromises = uniqueRequiredAuths?.map(async uAuth => {
-      const { accounts } = uAuth?.requiredAuthorization || {}
-      const accountsToGetSubAuths = accounts?.filter(
-        account => account.permission.permission !== Helpers.toChainEntityName('eosio.code'),
+      const accountsToGetSubAuths = uAuth?.requiredAuthorization?.accounts?.filter(
+        account => account.permission.permission !== EOS_IO_CODE,
       )
-      if (accountsToGetSubAuths?.length > 0) {
-        const accountsWithAuthPromises = accountsToGetSubAuths.map(async accObj => {
-          const { permission } = accObj
-          const { actor, permission: permissionName } = permission
-          const permToGetAuth: EosAuthorizationPerm = {
-            account: actor,
-            permission: permissionName,
-          }
-          const [subAuthPerm] = await this.addAuthToPermissions([permToGetAuth])
-          const { requiredAuthorization: subRequiredAuth } = subAuthPerm
-          if (!Helpers.isNullOrEmpty(subRequiredAuth?.accounts)) {
-            Errors.throwNewError('ChainJs doesnt support nested accounts permission')
-          }
-          return { ...accObj, subAuth: subRequiredAuth }
-        })
-        const accountsWithAuth = await Promise.all(accountsWithAuthPromises)
-        return { ...uAuth, requiredAuthorization: { ...uAuth.requiredAuthorization, accounts: accountsWithAuth } }
-      }
-      return uAuth
+      if (Helpers.isNullOrEmpty(accountsToGetSubAuths)) return uAuth
+      // if any authorization is itself an account, dig into it and get its auths (this is used for msig transactions)
+      const accountsWithAuthPromises = accountsToGetSubAuths.map(async accObj => {
+        const { permission } = accObj
+        const [permWithSubAuth] = await this.addAuthToPermissions([
+          { account: permission.actor, permission: permission.permission },
+        ])
+        const { requiredAuthorization: subRequiredAuth } = permWithSubAuth
+        if (!Helpers.isNullOrEmpty(subRequiredAuth?.accounts)) {
+          Errors.throwNewError('ChainJs doesnt support nested accounts permission')
+        }
+        return { ...accObj, subAuth: subRequiredAuth }
+      })
+      const accountsWithAuth = await Promise.all(accountsWithAuthPromises)
+      return { ...uAuth, requiredAuthorization: { ...uAuth.requiredAuthorization, accounts: accountsWithAuth } }
     })
+
     const uniqueRequiredAuthsWithSubAuth = await Promise.all(uniqueRequiredAuthsWithSubAuthPromises)
-    return uniqueRequiredAuthsWithSubAuth
+    this._requiredAuthorizationsWithSubAuths = uniqueRequiredAuthsWithSubAuth
```

Comment:
Earlier this method used to return [] containing requiredAuthsWithSubAuths but right now it doesn't return anything, This might impact the service - [getRequiredAuthorizationsForTransaction](https://github.com/API-market/oreid-service/blob/cefb9d4b65f3df0775aaf5e0fbb067cae93d9348/web/backend/services/chain/chainEos/eosTransaction.ts#L162) method

Reply:
Goog catch.
But service (and other code) should use the public, standard interface Transaction.requiredAuthorizations() - that's what it is there for. Please fix service to use the correct property.